libyul/AsmParser.cpp

```diff
@@ -192,13 +196,67 @@ std::optional<std::pair<std::string_view, SourceLocation>> Parser::parseSrcComme
 	langutil::SourceLocation const& _commentLocation
 )
 {
-	static std::regex const argsRegex = std::regex(
-		R"~~(^(-1|\d+):(-1|\d+):(-1|\d+)(?:\s+|$))~~"  // index and location, e.g.: 1:234:-1
-		R"~~(("(?:[^"\\]|\\.)*"?)?)~~",                // optional code snippet, e.g.: "string memory s = \"abc\";..."
-		std::regex_constants::ECMAScript | std::regex_constants::optimize
-	);
-	std::match_results<std::string_view::const_iterator> match;
-	if (!regex_search(_arguments.cbegin(), _arguments.cend(), match, argsRegex))
+	std::string const s_args (_arguments);
+	CharStream argumentStream (s_args, "");
+	Scanner scanner (argumentStream);
+
+	int sourceIndex { -1 };
+	int start { -1 };
+	int end { -1 };
+	bool indexOutOfRange { false };
+	std::optional<std::string_view> snippet { std::nullopt };
+	std::string_view tail { _arguments.substr(_arguments.size()) };
+	auto const parseIndex = [](Scanner& _scanner) -> std::tuple<std::optional<int>, bool>
+	{
+		int sgn = 1;
```

Comment:
```suggestion
		int sign = 1;
```
Just one more letter :)